# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: 984db15bc9e2
# Full Hash: 984db15bc9e2b5baa20bdc86baaa1165c381d326
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 11: Migrate ShowPersistentNotification to use IPC r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227696
# ==============================================================================

diff -r 622a66cdac2e -r 984db15bc9e2 dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Thu Nov 07 17:32:07 2024 +0000
+++ b/dom/notification/NotificationParent.cpp	Thu Nov 07 17:32:07 2024 +0000
@@ -62,6 +62,9 @@
     // XXX: QM_TRY?
     (void)NS_WARN_IF(NS_FAILED(UnpersistNotification(mPrincipal, mId)));
     (void)NS_WARN_IF(NS_FAILED(FireCloseEvent()));
+
+    // Unpersisted already and being unregistered already by nsIAlertsService
+    mDangling = true;
     Close();
 
     return NS_OK;